File: backend/src/models/AIConversation.js
const mongoose = require('mongoose');

const messageSchema = new mongoose.Schema(
  {
    role: { type: String, enum: ['user', 'assistant'], required: true },
    content: { type: String, required: true },
  },
  { _id: false, timestamps: true }
);

const aiConversationSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    title: { type: String, default: 'Cuộc hội thoại mới' },
    messages: [messageSchema],
    summary: { type: String }, // bản tóm tắt do Gemini tạo
  },
  { timestamps: true }
);

module.exports = mongoose.model('AIConversation', aiConversationSchema);

File: mobile/lib/models/ai_conversation.dart
class AIMessage {
  final String role; // 'user' | 'assistant'
  final String content;

  AIMessage({required this.role, required this.content});

  factory AIMessage.fromJson(Map<String, dynamic> json) {
    return AIMessage(role: json['role'] ?? 'user', content: json['content'] ?? '');
  }
}

class AIConversation {
  final String id;
  final String title;
  final String? summary;
  final List<AIMessage> messages;

  AIConversation({required this.id, required this.title, this.summary, this.messages = const []});

  factory AIConversation.fromJson(Map<String, dynamic> json) {
    return AIConversation(
      id: json['_id'] ?? '',
      title: json['title'] ?? 'Cuộc hội thoại',
      summary: json['summary'],
      messages: (json['messages'] as List<dynamic>? ?? [])
          .map((m) => AIMessage.fromJson(m as Map<String, dynamic>))
          .toList(),
    );
  }
}
